{{cookiecutter.project_slug}}/src/{{cookiecutter.module_name}}/version.py

```diff
@@ -1 +1,8 @@
-__version__ = '0.1.0'
+{% if cookiecutter.package_manager == 'poetry' %}try:
+    import importlib.metadata as importlib_metadata
+except ModuleNotFoundError:
+    import importlib_metadata
+
+__version__ = importlib_metadata.version("{{ cookiecutter.module_name }}")
```

Comment:
interesting approach to get the version number. I'm not a fan of having yet another dependency just for that task, but I'm also not sure what's the best alternative here.

[`importlib.metadata`](https://docs.python.org/3/library/importlib.metadata.html#distribution-versions) seems to require Python 3.7 or higher... using [`pkg_resources`](https://github.com/python-poetry/poetry/issues/273) could work, but is this even a part of the standard lib?

I just want to avoid adding too much complexity to this file. Not sure how to achieve that though 😄

Reply:
I think `pkg_resources` would have the same problem, because it is part of `setuptools` which is not part of the standard library. There are 3 solutions I can see:

- Simply require the python version to be >= 3.7 (IMO the best solution)
- The solution presented here
- Duplicate the version information

I'm happy to implement either of the others.

Reply:
well, I think we're already annoying some of our users by making Python 3.6 a requirement, so making 3.7 the lowest supported version probably won't help ;)
I don't know, duplicating the version is also not pretty.

Maybe let's leave it like this for now and keep looking for a better solution (or wait until we make the switch to Python 3.7 as min requirement)